PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../../environments/environment';
+import { HttpClient } from '@angular/common/http';
+import { ImageResponse } from "../../models/image/imageResponse";
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ImageService {
+
+  protected hostAddress = environment.hostAddress;
+
+  constructor(private httpClient: HttpClient) { }
+
+  post(file: File){
+    let formData = new FormData();
+    formData.append('file',file,file.name);
+    return this.httpClient.request<ImageResponse>('POST',this.hostAddress + "image",{body: formData});
```

Comment:
you can easily convert that to standard post request with location header

Reply:
with using abstract service

Reply:
Does not work with httpClient correctly.